hmy/api_backend.go

```diff
@@ -448,27 +448,15 @@ func (b *APIBackend) GetValidatorInformation(
 	// delete entry for previous epoch
 	b.apiCache.Forget(prevKey)
 
-	// calculate last APRHistoryLength epochs for averaging APR
-	epochFrom := bc.Config().StakingEpoch
-	nowMinus100 := now.Sub(now, big.NewInt(staking.APRHistoryLength))
-	if nowMinus100.Cmp(epochFrom) > 0 {
-		epochFrom = nowMinus100
-	}
-
+	// at this point, validator is active and has apr's for the recent 100 epochs
 	// compute average apr over history
 	if avgAPR, err := b.SingleFlightRequest(
 		key, func() (interface{}, error) {
 			total := numeric.ZeroDec()
 			count := 0
-			activated := false
-			for i := nowMinus100.Int64(); i < now.Int64(); i++ {
-				if apr, ok := stats.APRs[i]; ok {
-					total = total.Add(apr)
-					activated = true
-				}
-				if activated {
-					count = count + 1
-				}
+			for i := range stats.APRs {
```

Comment:
Why make this logic simpler now? Then we are not counting the empty gap Epochs anymore. And it will make APR higher for validators who drops out every now and then.

Reply:
We are accounting for those. This logic is only executed when validator is active. For active validator we would have already have entries for until last 100 epochs. So we just accumulate everything here.

Reply:
done.